host/class/uvc/usb_host_uvc_2/include/usb/uvc_host.h

```diff
@@ -0,0 +1,241 @@
+ *     - false: The frame was not yet processed by user.
+ *              The user must call uvc_host_frame_return() to return it to UVC driver.
+ */
+typedef bool (*uvc_host_frame_callback_t)(const uvc_host_frame_t *frame, void *user_ctx);
+
+/**
+ * @brief Configuration structure of UVC device
+ *
+ */
+typedef struct {
+    uvc_host_stream_callback_t event_cb;  /**< Stream's event callback function. Can be NULL */
+    uvc_host_frame_callback_t frame_cb;   /**< Stream's frame callback function */
+    void *user_ctx;                       /**< User's argument that will be passed to the callbacks */
+    struct {
+        uint16_t vid;                     /**< Device's Vendor ID. Set to 0 for any */
+        uint16_t pid;                     /**< Device's Product ID. Set to 0 for any */
+        uint8_t uvc_function_index;       /**< Index of UVC function you want to use. Set to 0 to use first available UVC function */
+    } usb;
+    uvc_host_stream_format_t vs_format;   /**< Video Stream format. Resolution, FPS and encoding */
+    struct {
+        int number_of_frame_buffers; /**< Number of frame buffers. These can be very large as they must hold the full frame.*/
+        size_t frame_size;           /**< 0: Use frame size from format negotiation result (might be too big).
+                                          (0; SIZE_MAX>: Use user provide frame size. */
+        uint32_t frame_heap_caps;    /**< Memory capabilities for frame buffers. Directly passed to heap_caps_malloc() */
+        int number_of_urbs;          /**< Number of URBs for this stream. Triple buffering scheme is recommended */
+        size_t urb_size;             /**< Size in bytes of 1 URB, 10kB should be enough for start.
+                                          Larger value results in less frequent interrupts at the cost of memory consumption */
+    } advanced;
+} uvc_host_stream_config_t;
+
+/**
+ * @brief Install UVC driver
+ *
+ * - USB Host Library must already be installed before calling this function (via usb_host_install())
+ * - This function should be called before calling any other UVC driver functions
+ *
+ * @param[in] driver_config Driver configuration structure. If set to NULL, a default configuration will be used.
+ * @return esp_err_t
+ */
+esp_err_t uvc_host_install(const uvc_host_driver_config_t *driver_config);
+
+/**
+ * @brief Uninstall UVC driver
+ *
+ * - User must ensure that all UVC devices must be closed via uvc_host_stream_close() before calling this function
+ *
+ * @return esp_err_t
+ */
+esp_err_t uvc_host_uninstall(void);
+
+/**
+ * @brief Open UVC compliant device
+ *
+ * @param[in]  stream_config  Configuration structure of the device
+ * @param[in]  timeout        Timeout in FreeRTOS ticks
+ * @param[out] stream_hdl_ret UVC stream handle
+ * @return esp_err_t
+ */
+esp_err_t uvc_host_stream_open(const uvc_host_stream_config_t *stream_config, int timeout, uvc_host_stream_hdl_t *stream_hdl_ret);
```

Comment:
Is that possible to open several streams on one physical device?

Reply:
Yes, this was also tested with a camera from a customer that offers 2 UVC interfaces in 1 USB deivce

Reply:
I have not tested this. Although, opening twice should not be possible.

Using the stream handle from multiple threads should be possible